crates/rpc-types-eth/src/block.rs

```diff
@@ -51,6 +52,32 @@ impl<T: TransactionResponse, H> Block<T, H> {
     }
 }
 
+impl<T> Block<T> {
+    /// Constructs an "uncle block" from the provided header.
+    ///
+    /// This function creates a new [`Block`] structure for uncle blocks (ommer blocks),
+    /// using the provided [`alloy_consensus::Header`].
+    pub fn uncle_block_from_header(header: alloy_consensus::Header) -> Self {
+        Self {
+            uncles: vec![],
+            header: header.clone().seal(header.hash_slow()).into(),
+            transactions: BlockTransactions::Uncle,
+            withdrawals: Some(vec![]),
+            size: Some(U256::from(
+                alloy_consensus::Block {
+                    header,
+                    body: alloy_consensus::BlockBody::<TxEnvelope> {
+                        transactions: vec![],
+                        ommers: vec![],
+                        withdrawals: None,
+                    },
+                }
+                .length(),
+            )),
```

Comment:
the size is also based on the entire block so we need to create the block first, then get it's size

Reply:
@mattsse Normally it's good. I just quickly implemented what you did in https://github.com/alloy-rs/alloy/pull/1561 (we can merge the two one after the other it should work) so that I can use it directly here. So the logic is for this line:
- Create a consensus primitive block
- Calculate its length
- Insert it here in the block size